crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -392,6 +392,30 @@ impl<'a> Resolver<'a> {
         }
 
         let span = path.span();
+        if let Some(type_alias_type) = self.lookup_type_alias(path.clone()) {
+            let mut args = vecmap(args, |arg| self.resolve_type_inner(arg, new_variables));
+            let expected_generic_count = type_alias_type.borrow().generics.len();
+
+            if args.len() != expected_generic_count {
+                self.push_err(ResolverError::IncorrectGenericCount {
+                    span,
+                    struct_type: type_alias_type.borrow().to_string(),
+                    actual: args.len(),
+                    expected: expected_generic_count,
+                });
+
+                // Fix the generic count so we can continue typechecking
+                args.resize_with(expected_generic_count, || Type::Error);
+            }
+
+            // Type Alias is monomorphized here because we do type checking before monomorphization
+            // If this is done in normal monomorphization phase type checking will return errors
+            // TODO(ethan): not sure if this is the best practice though
+            let typ = type_alias_type.borrow().get_type(&args);
```

Comment:
Defer to @jfecher on the best practice here. Should we just make a more explicit partial monomorphization pass for type aliases?

Reply:
This isn't monomorphization since it has nothing to do with generics. Aside from wording though the code itself is fine.